tests/v1/generation/conftest.py

```diff
@@ -0,0 +1,18 @@
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Copyright contributors to the vLLM project
+import os
+
+import pytest
+
+
+@pytest.fixture(autouse=True)
+def enable_batch_invariant_mode():
+    """Automatically enable batch invariant kernel overrides for all tests."""
+    old_value = os.environ.get("VLLM_BATCH_INVARIANT")
+    os.environ["VLLM_BATCH_INVARIANT"] = "1"
+    yield
+    # restore original value after test
+    if old_value is None:
+        os.environ.pop("VLLM_BATCH_INVARIANT", None)
+    else:
+        os.environ["VLLM_BATCH_INVARIANT"] = old_value
```

Comment:
Can we use monkeypatch instead? Also, why is this folder called generation if it is only for batch invariance? We should rename to relate to determinism

Reply:
Done